src/webgpu/api/operation/rendering/depth.spec.ts

```diff
@@ -3,13 +3,211 @@ Test related to depth buffer, depth op, compare func, etc.
 `;
 
 import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { TypedArrayBufferView } from '../../../../common/util/util.js';
 import { kDepthStencilFormats, kTextureFormatInfo } from '../../../capability_info.js';
 import { GPUTest } from '../../../gpu_test.js';
 
+class DepthTest extends GPUTest {
+  checkStencilCompareFunction(
+    compareFunction: GPUCompareFunction,
+    stencilColor: number[],
+    stencilRefValue: number,
+    stencilColorExpected: boolean
+  ) {
+    const depthSpencilFormat: GPUTextureFormat = 'depth24plus-stencil8';
+    const baseColor = [255, 255, 255, 255];
+
+    const baseStencilState = {
+      compare: 'always' as GPUCompareFunction,
+      failOp: 'keep' as GPUStencilOperation,
+      depthFailOp: 'keep' as GPUStencilOperation,
+      passOp: 'replace' as GPUStencilOperation,
+    };
+    const stencilState = {
+      compare: compareFunction,
+      failOp: 'keep' as GPUStencilOperation,
+      depthFailOp: 'keep' as GPUStencilOperation,
+      passOp: 'keep' as GPUStencilOperation,
+    };
+
+    const baseState = {
+      format: depthSpencilFormat,
+      depthWriteEnabled: false,
+      depthCompare: 'always' as GPUCompareFunction,
+      stencilFront: baseStencilState,
+      stencilBack: baseStencilState,
+    };
+
+    const state = {
+      format: depthSpencilFormat,
+      depthWriteEnabled: false,
+      depthCompare: 'always' as GPUCompareFunction,
+      stencilFront: stencilState,
+      stencilBack: stencilState,
+    };
+
+    this.runStencilStateTest(
+      baseState,
+      state,
+      stencilRefValue,
+      stencilColorExpected ? stencilColor : baseColor
+    );
+  }
+
+  runStencilStateTest(
+    baseState: GPUDepthStencilState,
+    state: GPUDepthStencilState,
+    stencilRefValue: number,
+    stencilColor: number[]
+  ) {
+    const renderTargetFormat = 'rgba8unorm';
+    const renderTarget = this.device.createTexture({
+      format: renderTargetFormat,
+      size: { width: 1, height: 1, depthOrArrayLayers: 1 },
+      usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.RENDER_ATTACHMENT,
+    });
+
+    const depthSpencilFormat: GPUTextureFormat = 'depth24plus-stencil8';
+    const depthTexture = this.device.createTexture({
+      size: { width: 1, height: 1, depthOrArrayLayers: 1 },
+      format: depthSpencilFormat,
+      sampleCount: 1,
+      mipLevelCount: 1,
+      usage: GPUTextureUsage.RENDER_ATTACHMENT | GPUTextureUsage.COPY_DST,
+    });
+
+    const depthStencilAttachment: GPURenderPassDepthStencilAttachment = {
+      view: depthTexture.createView(),
+      depthLoadOp: 'load',
+      depthStoreOp: 'store',
+    };
+
+    depthStencilAttachment.stencilLoadOp = 'load';
+    depthStencilAttachment.stencilStoreOp = 'store';
```

Comment:
nit: Why not just define these inline?

Reply:
Ah, just copied from the references. Move them to inline.